library/std/src/net/ip.rs

```diff
@@ -1884,10 +1896,16 @@ impl From<[u16; 8]> for IpAddr {
 // Tests for this module
 #[cfg(all(test, not(target_os = "emscripten")))]
 mod tests {
+    use crate::mem::size_of;
     use crate::net::test::{sa4, sa6, tsa};
     use crate::net::*;
     use crate::str::FromStr;
 
+    #[test]
+    fn ipv6bytes_all_fields_have_the_same_size() {
+        assert_eq!(size_of::<[u8; 16]>(), size_of::<[u16; 8]>());
```

Comment:
Shouldn't this be linked to the fields like `Ipv6Bytes::addr` and `Ipv6Bytes::addr16`? It may be possible that users set the union to some other types and this test case remains the same as it is currently but it was actually broken.

Reply:
Eh, it cannot be because addr and addr16 are not associated types. 
But I do have a way to check fields size, even though it contains unsafe.

Reply:
I think even using unsafe is better than leaving this hanging, having no relationship with the original union.

Reply:
Oops, I found that I don't even need unsafe.